### ISSUE TITLE
[Bug] Fail to release resources

### ISSUE THREAD

**Issue by gkevinyen5418** [hpcaitech/scalable-ml#7]
@Xu-Kai

It seems that sometimes we fail to release resources as the underlying scheduling strategy fails to acquire lock

https://github.com/hpcaitech/scalable-ml/blob/3d06076a94a15916d4ab5f871b32ef73fed38dea/cpp/scheduler/las_scheduler.cc#L254

thus the task scheduler stucks at this line waiting.

https://github.com/hpcaitech/scalable-ml/blob/3d06076a94a15916d4ab5f871b32ef73fed38dea/cpp/scheduler/task_scheduler.cc#L582

and the resources are not released.

**Comment by gkevinyen5418** [comment]
Process to reproduce.
1. Generate related files
```bash
git checkout feature/torch_vision_example
cd scalable-ml/example/torch_vision
python3 dispatch.py 60_job_processed.csv {DATA_PATH} file://{FILESTORE_PATH} 0 3 3
```
2. Execution
```bash
idev
bash exp.sh
```

**Comment by gkevinyen5418** [comment]
Fixed in [80172e5](https://github.com/hpcaitech/scalable-ml/commit/80172e5868fc99e31e56ea2a5317d1ecac2b54ef)
